cmd/erasure-multipart.go

```diff
@@ -993,14 +993,23 @@ func (er erasureObjects) CompleteMultipartUpload(ctx context.Context, bucket str
 	//
 	// Therefore, we adjust all ETags sent by the client to match what is stored
 	// on the backend.
-	kind, isEncrypted := crypto.IsEncrypted(fi.Metadata)
+	kind, _ := crypto.IsEncrypted(fi.Metadata)
 
 	var objectEncryptionKey []byte
-	if isEncrypted && kind == crypto.S3 {
+	switch kind {
+	case crypto.S3, crypto.S3KMS:
 		objectEncryptionKey, err = decryptObjectMeta(nil, bucket, object, fi.Metadata)
 		if err != nil {
 			return oi, err
 		}
+		if len(objectEncryptionKey) == 32 {
+			var key crypto.ObjectKey
+			copy(key[:], objectEncryptionKey)
+			opts.EncryptFn = metadataEncrypter(key)
+		}
+	case crypto.SSEC:
+		// TODO: Currently un-handled.
+		// We have not previously required SSE-C header to be sent on CompleteMultipartUpload.
```

Comment:
SSEC will never be sent in CompleteMultipart AFAIK @klauspost so not sure if we need to add this comment here.

Reply:
ok, we probably need to handle this case. Main problem is that we do not have the key to encrypt anything on CompleteMultipartUpload.

Reply:
wait @klauspost we actually do have it - it was simply never needed since we did not have to decrypt the ETag() however client sends the relevant headers so it must be passed down from the top layer.

If its not doing then we should fix it.

Reply:
This will make it easier to add: https://github.com/minio/minio/pull/16179/files#diff-205956958be9f6f40f7e0a8cac9e9d6fc9a7a1f155463f6c865d1dd22115fad9R1088